plugin/kubernetes/controller.go

```diff
@@ -226,6 +242,16 @@ func podListFunc(ctx context.Context, c kubernetes.Interface, ns string, s label
 	}
 }
 
+func endpointSliceListFunc(ctx context.Context, c kubernetes.Interface, ns string, s labels.Selector) func(meta.ListOptions) (runtime.Object, error) {
+	return func(opts meta.ListOptions) (runtime.Object, error) {
+		if s != nil {
+			opts.LabelSelector = s.String()
+		}
+		listV1, err := c.DiscoveryV1beta1().EndpointSlices(ns).List(ctx, opts)
```

Comment:
nit, should we just `return c.DiscoveryV1beta1().EndpointSlices(ns).List(ctx, opts)`

Reply:
Did it in the other ListFunc and WatchFunc Functions too (for Services, Pods, and Endpoints).